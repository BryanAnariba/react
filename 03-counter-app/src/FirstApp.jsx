import PropTypes from "prop-types";

export const FirstApp = ({title='No title!', subtitle='No Subtitle'}) => {
  return (
    <>
      <h1 data-testid="test-title">{title}</h1>
      <p>{subtitle}</p>
      <p>{subtitle}</p>
    </>
  );
};

FirstApp.propTypes = {
  title: PropTypes.string,
  subtitle: PropTypes.string,
};

// Esto esta Deprecated
// FirstApp.defaultProps = {
//     title: 'No title!',
//     subtitle: 'Not Subtitle!'
// };
