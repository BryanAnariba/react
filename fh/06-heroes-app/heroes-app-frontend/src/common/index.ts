export * from './routers';
export * from './components';
export * from './hooks';