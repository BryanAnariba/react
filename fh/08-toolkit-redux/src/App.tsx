import reactLogo from "./assets/react.svg";
import { useDispatch, useSelector } from "react-redux";
import { decrement, increment, incrementBy, RootState } from "./store";
import "./App.css";

function App(): JSX.Element {
  const counter = useSelector((state: RootState) => state.counter.counter);
  const dispatch = useDispatch();

  return (
    <>
      <div>
        <a href="https://react.dev" target="_blank">
          <img src={reactLogo} className="logo react" alt="React logo" />
        </a>
      </div>
      <h1>{counter}</h1>
      <div className="card">
        <button onClick={() => dispatch(increment())} style={{marginLeft: 5}}>
          + 1
        </button>
        <button onClick={() => dispatch(decrement())} style={{marginLeft: 5}}>
          - 1
        </button>
        <button onClick={() => dispatch(incrementBy(2))} style={{marginLeft: 5}}>
          + 2
        </button>
      </div>
    </>
  );
}

export default App;
