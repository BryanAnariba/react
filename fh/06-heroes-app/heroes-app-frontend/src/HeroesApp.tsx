import { AppRouter } from "./common"
import { AuthProvider } from "./modules/auth"

export const HeroesApp = (): JSX.Element => {
  return (
    <>
      <AuthProvider>
        <AppRouter />
      </AuthProvider>
    </>
  )
}
