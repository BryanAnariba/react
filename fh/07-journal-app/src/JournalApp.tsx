import { AppRouter, AppTheme } from "./common";

export const JournalApp = (): JSX.Element => {
  return (
    <AppTheme>
      <AppRouter />
    </AppTheme>
  );
};
