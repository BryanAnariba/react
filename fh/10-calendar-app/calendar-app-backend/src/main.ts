import { NestFactory } from '@nestjs/core';
import { AppModule } from './app.module';
import { Logger, ValidationPipe } from '@nestjs/common';
import { envs } from './common/config';

async function bootstrap() {
  const logger = new Logger('Calendar App');

  const app = await NestFactory.create(AppModule);
  app.setGlobalPrefix('api');

  app.enableCors();

  app.useGlobalPipes(
    new ValidationPipe({
      transform: true,
      forbidNonWhitelisted: true,
    })
  );

  await app.listen(envs.port);
  
  logger.log(`===> Calendar App Backend is running on port: ${envs.port} <===`);
}
bootstrap();
