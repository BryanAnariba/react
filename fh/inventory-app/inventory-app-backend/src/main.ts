import { NestFactory } from '@nestjs/core';
import { AppModule } from './app.module';
import { Logger, ValidationPipe } from '@nestjs/common';
import { envVariables } from './inventory/core/config';

async function bootstrap() {
  const logger = new Logger('Inventory App');

  const app = await NestFactory.create(AppModule);

  app.setGlobalPrefix('api');

  app.enableCors();

  app.useGlobalPipes(new ValidationPipe({
    transform: true,
    whitelist: true,
    forbidNonWhitelisted: true,
  }));

  await app.listen(envVariables.port);

  logger.log(`===>Inventory App running on port: ${envVariables.port}<===`);
}
bootstrap();
