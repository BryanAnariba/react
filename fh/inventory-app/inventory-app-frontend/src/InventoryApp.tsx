import { Provider } from 'react-redux';
import { store } from './store';
import { BrowserRouter } from 'react-router';
import { AppRouter } from './router/AppRouter';
import './App.css';

function InventoryApp (): JSX.Element {
  return (
    <Provider store={store}>
      <BrowserRouter>
        <AppRouter />
      </BrowserRouter>
    </Provider>
  )
}

export default InventoryApp;
