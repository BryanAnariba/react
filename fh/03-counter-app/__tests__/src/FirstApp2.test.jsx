import { render, screen } from "@testing-library/react";
import { FirstApp } from "../../src/FirstApp";

describe('Testing in ./src/FirstApp.jsx', () => {
  const title = 'Hello i am Goku Gomez';
  const subtitle = 'A caso con el pelo se estudia!';

  test('Should to be match with the snapshot', () => {
    const { container } = render(<FirstApp title={title} subtitle={subtitle} />);
    expect(container).toMatchSnapshot();
  });

  test('Should to show the message "Hello i am Goku Gomez"', () => {
    render(<FirstApp title={title} subtitle={subtitle} />);
    expect(screen.getByText(title)).toBeTruthy();
  });

  test('Should to be show the title in h1 tag', () => {
    render(<FirstApp title={title} subtitle={subtitle} />);
    expect(screen.getByRole('heading', { level: 1 }).innerHTML).toContain(title);
  });

  test('Should to be show the subtitle sent by props', () => {
    render(<FirstApp title={title} subtitle={subtitle} />);
    expect(screen.getAllByText(subtitle).length).toBe(2);
  });
});