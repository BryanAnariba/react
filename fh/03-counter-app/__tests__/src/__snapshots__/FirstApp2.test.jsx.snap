// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing in ./src/FirstApp.jsx Should to be match with the snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hello i am Goku Gomez
  </h1>
  <p>
    A caso con el pelo se estudia!
  </p>
  <p>
    A caso con el pelo se estudia!
  </p>
</div>
`;
